Userland/Libraries/LibCore/EventLoop.h

```diff
@@ -25,6 +25,8 @@
 
 namespace Core {
 
+// NOTE: If you came here due to an initializiation-order fiasco, you probably created a global event loop variable.
+//       That is a bad idea and you should stop doing what you're doing right now.
```

Comment:
Can we catch this programmatically (at runtime) instead perhaps?
Maybe LibC could have an internal API that tells you if all init-time constructors have executed or something?

Reply:
Compile-time seems infeasible (a clang plugin is possible but only for clang + toolchain rebuilds for everybody). I have no idea how to do this kind of checking in LibC but I'll have a look.